tests/opower/utilities/test_smud.py

```diff
@@ -0,0 +1,96 @@
+"""Tests for SMUD."""
+import os
+import unittest
+
+import aiohttp
+from dotenv import load_dotenv
+
+from src.opower.utilities.smud import (
+    SMUD,
+    SMUDLoginParser,
+    SMUDOktaResponseSamlResponseValueParser,
+)
+
+REQUEST_VERIFICATION_TOKEN_HTML_FILENAME = os.path.join(
+    os.path.dirname(__file__), "smud/myAccountResponse.html"
+)
+OKTA_SAML_RESPONSE_HTML_FILENAME = os.path.join(
+    os.path.dirname(__file__), "smud/smudOkta.html"
+)
+ENV_SECRET_PATH = os.path.join(os.path.dirname(__file__), "../../../.env.secret")
+
+
+class TestSMUD(unittest.IsolatedAsyncioTestCase):
+    """Test public methods inherited from UtilityBase."""
+
+    def test_name(self):
+        """Test name."""
+        smud = SMUD()
+
+        self.assertEqual("Sacramento Municipal Utility District (SMUD)", smud.name())
+
+    def test_subdomain(self):
+        """Test subdomain."""
+        smud = SMUD()
+
+        self.assertEqual("smud", smud.subdomain())
+
+    def test_timezone(self):
+        """Test timezone."""
+        smud = SMUD()
+
+        self.assertEqual("America/Los_Angeles", smud.timezone())
+
+    async def test_real_login(self):
+        """Perform a live test against the SMUD, OKTA and Opower websites."""
+        if os.path.exists(ENV_SECRET_PATH) is False:
+            self.skipTest(
+                "Add `SMUD_USERNAME=` and `SMUD_PASSWORD=` to `.env.secret` to run live SMUD test."
+            )
+
+        load_dotenv(dotenv_path=ENV_SECRET_PATH)
+
+        username = os.getenv("SMUD_USERNAME")
+        password = os.getenv("SMUD_PASSWORD")
+
+        if username is None or password is None:
+            self.skipTest(
+                "Add `SMUD_USERNAME=` and `SMUD_PASSWORD=` to `.env.secret` to run live SMUD test."
+            )
+
+        smud = SMUD()
+        session = aiohttp.ClientSession()
```

Comment:
```suggestion
        session = aiohttp.ClientSession()
        self.addCleanup(session.close)
```
This would avoid unrelated lengthy messages along the lines of
```
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x106443710>
source_traceback: Object created at (most recent call last):
```
on test failure

Reply:
:P if you want to play with async you can also do 
```
    async with aiohttp.ClientSession() as session:
        smud = SMUD()
        await smud.async_login(session, username, password, None)
``` but for simplicity @v-yarotsky wins.